WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -48,6 +51,9 @@ def create
         @competition.errors[:competition_id_to_clone] = "invalid"
       end
     end
+    if current_user.any_kind_of_delegate?
+      @competition.delegates << current_user
+    end
 
     if @competition.errors.size == 0 && @competition.save
```

Comment:
Oh, totally missed this. :sob: 

Moving it to the competition model sounds reasonable. I'm not sure how right now, though.